civicrm_resources/civicrm_resources.module

```diff
@@ -6,205 +6,43 @@
-        continue;
-      }
-    }
-  }
-
-  foreach ($css_list as $key => $css_file) {
-    drupal_add_css(ltrim(str_replace(DRUPAL_ROOT, '', $css_file), '/'));
-  }
-  foreach ($js_list as $key => $js_file) {
-    drupal_add_js(ltrim(str_replace(DRUPAL_ROOT, '', $js_file), '/'),  array('scope' => 'footer'));
-  }
-}
-
-/**
- * Filters files into css or js
- *
- * @param array $files
- *
- * @return array
- */
-function _civicrm_resources_filter_files($files) {
-  $filtered_js_file = array_filter($files, function ($file) {
-    return preg_match('/\.js$/', $file['filename']);
-  });
-
-  $filtered_css_file = array_filter($files, function ($file) {
-    return preg_match('/\.css$/', $file['filename']);
-  });
-
-  return [$filtered_js_file, $filtered_css_file];
-}
-
-/**
- * Retrieves uri for file list
- *
- * @param array $files
- *
- * @return array
- */
-function _civicrm_resources_get_uris($files) {
-  $resource_uri = array_map(function ($file) {
-    return  $file->uri;
-  }, $files);
-
-  return array_values($resource_uri);
-}
-
-/**
- * Retrieves uri for a specific file
- *
- * @param int $index
- * @param array $resource_list
- *
- * @return string
- */
-function _civicrm_resources_get_file_uri($index, $resource_list) {
-  $uri_values = array_values($resource_list);
-
-  return $uri_values[$index]['uri'];
-}
-/**
- * Finds all files that match a given mask in a given directory.
- *
- * Directories and files beginning with a period are excluded; this
- * prevents hidden files and directories (such as SVN working directories)
- * from being scanned.
- *
- * @param $dir
- *   The base directory or URI to scan, without trailing slash.
- * @param $mask
- *   The preg_match() regular expression of the files to find.
- * @param $options
- *   An associative array of additional options, with the following elements:
- *   - 'nomask': The preg_match() regular expression of the files to ignore.
- *     Defaults to '/(\.\.?|CVS)$/'.
- *   - 'callback': The callback function to call for each match. There is no
- *     default callback.
- *   - 'recurse': When TRUE, the directory scan will recurse the entire tree
- *     starting at the provided directory. Defaults to TRUE.
- *   - 'key': The key to be used for the returned associative array of files.
- *     Possible values are 'uri', for the file's URI; 'filename', for the
- *     basename of the file; and 'name' for the name of the file without the
- *     extension. Defaults to 'uri'.
- *   - 'min_depth': Minimum depth of directories to return files from. Defaults
- *     to 0.
- * @param $depth
- *   Current depth of recursion. This parameter is only used internally and
- *   should not be passed in.
- *
- * @return
- *   An associative array (keyed on the chosen key) of objects with 'uri',
- *   'filename', and 'name' members corresponding to the matching files.
+ * @param string $extension_path
+ *   Machine path of civicrm extension.
```

Comment:
This is not really a path. In CiviCRM this is called `extension key`. Could you please update the param to use this terminology?

Reply:
Are you sure @davialexandre ? Because it actually is an absolute path to the extension folder.